### PR TITLE
Add search provider tests

### PR DESCRIPTION
**Pull request by efritz**
Add unit tests to search providers. Require refactoring of API functions, which are now in a mockable class.

### CODE REVIEW

File: shared/util/graphql.ts [hunk 1]
```diff
@@ -49,3 +49,8 @@ function aggregateErrors(errors: Error[]): Error {
         errors,
     })
 }
+
+export const gql = (
+    template: TemplateStringsArray,
+    ...substitutions: any[]
+): string => String.raw(template, ...substitutions)
```
Comment (felixfbecker) [inline shared/util/graphql.ts]: FYI this is not the same as `tagged-template-noop` and actually wrong :(
`String.raw` prevents any escape sequences in the string, which is not the intention (the intention is to just be a noop, like `tagged-template-noop`). I wrote this in the main repo, realized later that it was wrong, but changing would mean all the queries need to be tested again. In all other repos I used `tagged-template-noop` which I only discovered after.

Comment (efritz) [reply shared/util/graphql.ts]: I'm having issues including that due to a tsconfig issue. Same with lru-cache. When I run the tests when it's imported I get

`TypeError: lru_cache_1.default is not a constructor`

what's the resolution for these kinds of issues?

Comment (efritz) [reply shared/util/graphql.ts]: Solved it!

Comment (felixfbecker) [reply shared/util/graphql.ts]: Glad you did. For visibility: The resolution is `esModuleInterop: true` in tsconfig.json, and making sure the right import style is used: `import foo from 'bar'` imports the _default export_, `import * as foo from 'bar'` import _everything_ as a module record object (which can never be a function).

**Pull request merged**